### DISCUSSION TITLE
Extending a 2D UnstructuredGrid to 3D

### DISCUSSION THREAD

**Discussion by pilarct** [pyvista/pyvista#3046] [Q&A]
Hi!
I would like to know if it is possible to extend two 2D UnstructuredGrid into a 3D? Here, an example of the meshes I would like to extend:

![2Dmeshes](https://user-images.githubusercontent.com/109792282/180408689-9a2b08b6-fe05-4699-9bae-cbcb24199938.png)

Thanks in advance!

**Comment by akaszynski** [comment]
Looks like you'd like to perform a 3D delaunay tessellation. You could try [delaunay_3d](https://docs.pyvista.org/api/core/_autosummary/pyvista.UnstructuredGridFilters.delaunay_3d.html), but it's not one of VTK/PyVista's strong suits.

**Comment by pilarct** [comment]
Hi!
I tried already with delaunay_3d, however it also meshes what is below the bottom curve like this:
<img src="https://user-images.githubusercontent.com/109792282/180783516-88807f27-a9af-4a3b-aba1-1ed84976726a.png" width="500">

And this is what I got using what is explained in here: https://docs.pyvista.org/examples/00-load/create-structured-surface.html#sphx-glr-examples-00-load-create-structured-surface-py
<img src="https://user-images.githubusercontent.com/109792282/180786434-84f904c9-9461-4acf-933b-142bc3b070ca.png" height="300"> <img src="https://user-images.githubusercontent.com/109792282/180789299-97e8d89c-bf59-413f-9ecf-804664601f16.png" height="300">

The problem I have with the structuredGrid is that I cannot close the structure because one edge would be duplicated.

**Comment by whophil** [comment]
I'm not sure I fully understand the question, but [extrude_trim](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.extrude_trim.html#pyvista.PolyDataFilters.extrude_trim) could be useful.
